crates/sui-adapter/src/execution_engine.rs

```diff
@@ -191,7 +191,11 @@ fn execute_transaction<
         execution_result
     });
     if !gas_status.is_unmetered() {
-        temporary_store.charge_gas(gas_object_ref.0, &mut gas_status, &mut result, gas);
+        if protocol_config.gas_checks_v2() {
```

Comment:
For my own clarity: This is just for testnet? i.e. there will be no versioning on `main`?

Reply:
correct